python/cudf_polars/cudf_polars/utils/config.py

```diff
@@ -188,6 +203,13 @@ class ShuffleMethod(str, enum.Enum):
     _RAPIDSMPF_SINGLE = "rapidsmpf-single"
 
 
+class MemoryResourceType(str, enum.Enum):
```

Comment:
Non-blocker, and I'll post more in https://github.com/rapidsai/cudf/issues/20475, but I think we should remove this in a future PR. I think that a declaration to use async or managed memory should apply regardless of what cluster / runtime you're using, and that should be feasible to implement.

Similar story for the other `client`-specific configuration options added here.